### PR TITLE
[DOC] Improve NEWS.md rendering and add missing features

### PR DESCRIPTION
**Pull request by zverok**
* Fix rendering of the file on [docs.ruby-lang.org](https://docs.ruby-lang.org/en/master/NEWS_md.html):
  * More indentation for code;
  * Properly qualified methods to link to docs (`Queue#freeze` → `Thread::Queue#freeze`, remove unnecessary backticks, etc.)
* Add two minor yet notable changes:
  * `Time.new` stricter parsing of strings;
  * `NoMethodError` rendering logic change.

### CODE REVIEW

File: NEWS.md
Comment (jeremyevans) [inline NEWS.md:71]: I don't think this change is worth mentioning in NEWS.

File: NEWS.md
Comment (jeremyevans) [inline NEWS.md:514]: When removing the NoMethodError change, we would want to remove this line as well.

**Review by jeremyevans — CHANGES_REQUESTED** [review]
Looks mostly good.  Only a couple changes requested.

File: NEWS.md
Comment (nobu) [reply NEWS.md:514]: It's better to sort by URLs.

File: NEWS.md
Comment (ko1) [inline NEWS.md:78]: why do you remove backquote even though it is `.md`?

File: NEWS.md
Comment (mame) [reply NEWS.md:71]: I think it is better to explain this in "Compatibility issues" section than in "Core classes updates".

File: NEWS.md
Comment (nobu) [reply NEWS.md:78]: It will be marked up as `<code>` without backquotes.

File: NEWS.md
Comment (zverok) [reply NEWS.md:71]: @jeremyevans My thinking here is this:
1. It is a part of public behavior
2. It is a behavior that is frequently encountered (i.e., almost every Rubyist sees this error almost every day, while some other notable changes like `WeakKeyMap` or `Fiber#kill` are something that wouldn't be used by as many people)
3. The change affects the quality of life. I understand the justification, yet the change might make somebody's debugging harder (by not seeing the real problematic object) and somebody's easier (by removing 20-lines `#inspect`s of complicated objects)
4. The change also might affect application/library code (this is purely theoretical, but not unimaginable that some code caught the error and extracted the `#inspect` of the affected object; or that somebody's `#inspect` was defined in the way that considered frequent rendering in this message)
5. The entry in `NEWS` not only explains the change but links to the corresponding discussion, so when one sees in the new version, "wait, is this changed?" they can a) find the entry in NEWS (and make sure the change was conscious, not some side-effect or bug) and b) find the justification.

File: NEWS.md
Comment (zverok) [reply NEWS.md:78]: @ko1 Without backticks RDoc is able to linkify the method name. 
The current version:
image
After the fix:
image

Comment (ko1) [reply NEWS.md:78]: I see.

----

For me, it is better to allow backquotes and rdoc also accepts backquoted method names, because it is easy to copy & paste to other articles.

Comment (nobu) [reply NEWS.md:78]: You mean the case of "only method name enclosed in backquotes", right?
E.g., `` `ObjectSpace::WeakMap#delete` ``, but not `` `(ObjectSpace::WeakMap#delete)` ``.
It would be better but should be requested to ruby/rdoc, not here.

File: NEWS.md
Comment (jeremyevans) [reply NEWS.md:71]: @zverok I read what you wrote and still do not think this is worth mentioning.  There is a tradeoff in mentioning anything in NEWS.  We want NEWS to be short enough that people will read it, hitting the important parts of the release.  NEWS is not designed to be comprehensive and discuss every single change since the last release. If you think this is a compatibility issue (I do not), take @mame's advice and put it in the compatibility issues section. Otherwise, I think it should be removed.

File: NEWS.md
Comment (jeremyevans) [inline NEWS.md:130]: This uses the same code twice, which doesn't appear intentional.  I would remove the duplicate lines.

**Review by jeremyevans — CHANGES_REQUESTED** [review]
One additional change requested.  After that, OK to merge without further review.

File: NEWS.md
Comment (zverok) [reply NEWS.md:130]: Ugh. Meant to show "year-month" and "year-month-day", it is a typo. But anyway, I think one example is enough.

**Review by jeremyevans — APPROVED** [review]
I did mention in my previous review that this could be merged without further review after the change I requested was made.

Please squash when merging.

File: NEWS.md
Comment (nurse) [reply NEWS.md]: I'm for because it is a part of public behavior.
Generally speaking NEWS.md is allowed to write about something if a committer thinks it should be.
I think what jeremy says  is rather a release note.

**Pull request merged**
